Packages/com.chisel.core/Chisel/Core/GeneratorBase/BrushFactory.Utility.cs

```diff
@@ -1233,5 +1233,47 @@ public static void CreateExtrudedSubMesh(ref BrushMesh brushMesh, Vector3[] side
             if (!brushMesh.Validate(logErrors: true))
                 brushMesh.Clear();
         }
+
+        /// <summary>
+        /// Creates a brush out of a set of planes. A brush consists of several faces, each of which can be defined by a plane (of which there must be at least four).
+        /// Each plane defines a half space, that is, an infinite set of points that is bounded by a plane. The intersection of these half spaces forms a convex polyhedron.
+        /// The brush geometry is not centered unless your planes are, use <seealso cref="BrushMesh.CenterAndSnapPlanes"/> to center and position the game object to match.
+        /// <para>Note: The current algorithm only works within a limited range of +-4096 world units from the world center. If possible, perform your operations near the center of the world for optimal accuracy.</para>
+        /// </summary>
+        public static void CreateFromPlanes(ref BrushMesh brushMesh, float4[] planes, in ChiselSurfaceDefinition surfaceDefinition)
+        {
+            Debug.Assert(planes != null && planes.Length >= 4);
+
+            // TODO: check if we can cut up a box with infinite dimensions to get around the size constraint.
+
+            // build a very large cube brush.
+            surfaceDefinition.EnsureSize(6);
+            CreateBox(ref brushMesh, new Vector3(-4096, -4096, -4096), new Vector3(4096, 4096, 4096), in surfaceDefinition);
+
+            // cut the brush using the given planes.
+            brushMesh.Cut(planes);
+        }
+
+        /// <summary>
+        /// Creates a brush out of a set of points. A convex hull is generated that encompasses all points (of which there must be at least four).
+        /// The brush geometry is not centered unless your points are, use <seealso cref="BrushMesh.CenterAndSnapPlanes"/> to center and position the game object to match.
+        /// <para>Note: The current algorithm only works within a limited range of +-4096 world units from the world center. If possible, perform your operations near the center of the world for optimal accuracy.</para>
+        /// </summary>
+        public static void CreateFromPoints(ref BrushMesh brushMesh, Vector3[] points, in ChiselSurfaceDefinition surfaceDefinition)
+        {
+            Debug.Assert(points != null && points.Length >= 4);
+
+            // guess a capacity that's large enough to hold the planes, division by 6 due to triangulation.
+            // the convex hull has duplicate polygons, degenerate triangles and T-Junctions.
+            // as such this should be more than we will need to store a handful of unique planes.
+            var planes = new List<float4>(points.Length / 6);
+
+            // calculate a convex hull out of the points.
+            ConvexHullCalculator convexHullCalculator = new ConvexHullCalculator();
+            convexHullCalculator.GenerateHull(points, ref planes);
```

Comment:
... or just take the bounds of all the points used in the convex hull

Reply:
While the ConvexHullCalculator creates planes describing a convex hull, I made it efficiently calculate and output the resulting bounds. This is then passed into the CreateFromPlanes() function.